design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+
+Existing non-List data that follows List comma delimiter format will be in Text data type if `Allow List` has been turned off. Users can simply convert the non-List data into List data by turning on `Allow List`.
+
+### 4.1 Converting Correctly Formatted Non-List to List
+
+Correctly formatted Lists can be in non-List data types only when `Allow List` is turned off. As mentioned above, users can convert non-List data that follows List comma delimiter format by simply turning on `Allow List`. Mathesar can automatically parse the data into List and make inference for the List item data type. Users can change List item data type by following 2.2.
+
+---
+Wireframes
+
+[Converting Correctly Formatted Non-List to List](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.28&scaling=min-zoom&starting-point-node-id=8398%3A88843&show-proto-sidebar=1)
+
+---
+
+### 4.2 Converting Incorrectly Formatted Non-List to List
+
+Incorrectly formatted data cannot be automatically turned into List even when `Allow List` is turned on. However, this scenario will only happen when no comma is detected in one or more cells of the column (see 4.3). Users are informed in the following two ways:
+
+1. Tooltip will show up as soon as users hover over the information icon next to `Allow List` option. The tooltip aims to concisely convey the correct format of List data and point users to relevant help pages for details.
+2. When users turn on `Allow List` and Mathesar is unable to detect a correct List data format, a warning modal will show up. The warning should only show up when Mathesar detects certain List conversion intents from the users. For example, if users only have a single number in each cell, turning on `Allow List` is just to enable this feature and clearly the data is not yet in List format.
+
+---
+Wireframes
+
+[Converting Incorrectly Formatted Non-List to List](https://www.figma.com/proto/Uaf1ntcldzK2U41Jhw6vS2/Mathesar-MVP?node-id=7317%3A80859&viewport=306%2C48%2C0.37&scaling=min-zoom&starting-point-node-id=8399%3A90654&show-proto-sidebar=1)
+
+---
+
+### 4.3 Converting Partially Correctly Formatted Non-List to List
+
+If Mathesar uses comma delimiter to detect List, Mathesar can still successfully parse Text data type as long as a comma is present. In this case, if the resulting List items are inferred as an incorrect basic data type due to the data error (e.g., Mathesar is not successful in inferring Number due to a data error), users can quickly realize something is wrong. Unfortunately, users still need to locate the errors by themselves.
+
+## 5. Other Considerations
+
+The List data type design exploration above has assumed a fair amount of background processing capability in Mathesar. For example, the `Allow List` option can simplify users' decision-making process by detecting the existence of a List in the background constantly, as users are actively editing data. 
```

Comment:
Although we _can_ do this, I don't think we should be changing the structure of data without an active request from users to do so. Reasoning:
- Changing the structure of data may result in data loss, which is not something we should risk without explicit instructions from the user.
- Changing the structure of data also constrains what kind of data can be stored in a column.  

Unlike a spreadsheet, Mathesar requires that all data in a column have the same data type, and we cannot deal with inconsistencies well. This is a benefit of Mathesar in many use cases – we can guarantee data integrity, but that means we can't be as flexible.